OpenRA.Mods.Common/Widgets/ScrollPanelWidget.cs

```diff
@@ -167,7 +167,7 @@ public override void DrawOuter()
 			Game.Renderer.EnableScissor(drawBounds);
 
 			drawBounds.Offset((-ChildOrigin).ToPoint());
-			foreach (var child in Children)
+			foreach (var child in Children.ToArray())
```

Comment:
Did you add this because the event handlers are being raised on the listening thread and not the UI thread? If so I think you'll to better to invoke UI code on the UI thread or you will likely run into other threading nastiness.